### ISSUE TITLE
idea for margin-notes on plot

### ISSUE THREAD

**Issue by dankelley** [dankelley/mooring#25]
Now that I'm trying real-world moorings, I see that it's going to be very common to have labels overwriting each other (see below, from #24).  I think I will implement something I do a lot in things like scholarship rankings, which is to put a more extensive listing of elements over on the RHS, equispaced over the plot height, with lines pointing back to the actual mooring. 

I won't bother explaining this too much, since I'm not seeking advice or help on this. The purpose of this issue is just to be a component of a "to do" list.  Also, I will need to try this to see if it will work reasonably well, and I don't want to get into a tangle with a whole lot of extra arguments to control spacing, etc.  So ... I may do this, and I may drop the idea.

<details>

**messy diagram**

<img width="693" alt="Screen Shot 2021-06-09 at 8 25 44 AM" src="https://user-images.githubusercontent.com/99469/121346296-54813600-c8fc-11eb-87a1-5942de6b9768.png">

**similar diagram that's less messy**

I've cut off the text on the RHS (person names). Notice, though, that if I put that material alongside the scores (the thing plotted on the axis) then the text would be difficult to read.

<img width="462" alt="Screen Shot 2021-06-09 at 8 29 01 AM" src="https://user-images.githubusercontent.com/99469/121346865-fef95900-c8fc-11eb-880e-1a221fb8bb0f.png">

**Comment by dankelley** [comment]
Below is a rough proof-of-concept (not committed), just to see if text might fit.  I think it will.  Next steps:
* [ ] remove the red circles and instead draw lines back to the mooring trace
* [ ] invent an argument structure that will yield somewhat-reasonable results by default, with the ability for the user to control placement, font size, etc. A simple way to do that is to extend the handling of my new `showDetails` arg, to allow not just logicals but also lists.  If it's a list, then we could have like `showDetails=list(at=X, cex=CEX)` and so forth, so the user could set where they want the labels to start, and what character-expansion factor to use.

I'll think about this during my run this morning.  Oh man ... lovely sun and no humidity!!!!

![image](https://user-images.githubusercontent.com/99469/121536863-d1361200-c9d9-11eb-92e9-78762377ac0c.png)

**Comment by dankelley** [comment]
Oh, I see.  I don't have a `ballast` subclass -- that's the hickup.  (I was defining one via
```R
connector("ballast", 100/2.2, height=1, width=0.05, CD=1) # guess
```
but my code to display things on the RHS is not showing the locations of connectors (yet).

**Comment by dankelley** [comment]
I fiddled a bit more and have as follows.  There's something weird going on at the swivel/ballast location (notice the weird little jag to the left, in the left-hand panel a bit below 200m depth), and I'll look into that tomorrow.  (I had the ballast being buoyant until a minute ago -- that's a sign that the code is all pretty plastic.). And I have no idea whether I've even typed in the elements from the mooring diagram; for some reason, I seem to be reading '6' as '8' unless I zoom the diagram, etc -- wrong time of day to look at small things.

Anyway, below is a bit of an idea of how things might look with these "margin notes".  I like them, and so the decision of whether to add them has been made.  (None of this is pushed to GH, since there are no users as of yet, to my knowledge.)

![mooring_01](https://user-images.githubusercontent.com/99469/121737592-754db500-cacf-11eb-8c38-96589102e801.png)

**Comment by dankelley** [comment]
Actually, I don't think the horizontal jig-jags are an error.  The x axis labelling was throwing me off.  (Don't ask me why -- I wrote the label just hours before being thrown off.). The thing on the x axis is the *perturbation* from the normal depth of a given element.  There's no reason why that has to be smooth. For example, adding 100 pounds of ballast (element 4, counting from the top) would very likely pull that element down a bit.  And a float will very likely be pulled up a bit.

This is a bit hard to explain in words, and these are just notes to myself, so I won't bother trying to be articulate.  The new labelling, shown below, seems to me to be less confusing.  I chose to write "depth increase" rather than "depth perturbation" because this will always be an increase, for non-stretchy lines.  (In any case, if we had stretchy lines, the label would still make sense, with negative values appearing.)

With this, it's time to close this issue.  We now *have* the requested feature, and any improvements ought to be in new issues.

![mooring_01](https://user-images.githubusercontent.com/99469/121772990-6dcdf080-cb4f-11eb-80e0-5c0dfe1a6d3e.png)
